pallets/nfts/src/features/approvals.rs

```diff
@@ -153,6 +158,14 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		collection: T::CollectionId,
 		item: T::ItemId,
 	) -> DispatchResult {
+		let collection_details =
+			Collection::<T, I>::get(collection).ok_or(Error::<T, I>::UnknownCollection)?;
+
+		ensure!(
+			AccountApprovals::<T, I>::get(collection, collection_details.owner) == 0,
```

Comment:
Explanation: Throw if there is a non-zero number of collection approvals that the collection owner granted. Expect the user to remove all collection approvals first before clearing all transfer approvals of a specific item.